x-pack/plugins/infra/public/hooks/use_lens_attributes.ts

```diff
@@ -37,13 +37,18 @@ export const useLensAttributes = ({
   } = useKibana<InfraClientSetupDeps>();
   const { navigateToPrefilledEditor } = lens;
   const [formulaAPI, setFormulaAPI] = useState<FormulaPublicApi>();
+  const [error, setError] = useState<boolean>(false);
 
   useEffect(() => {
-    const getFormulaApi = async () => {
-      const { formula } = await lens.stateHelperApi();
-      return setFormulaAPI(formula);
-    };
-    getFormulaApi();
+    lens
+      .stateHelperApi()
+      .then(({ formula }) => {
+        setError(false);
+        setFormulaAPI(formula);
+      })
+      .catch(() => {
+        setError(true);
+      });
   }, [lens]);
```

Comment:
**suggestion(non-blocking):** a last small improvement in terms of readability could be to reuse the already installed react-use library which exports the useful `useAsync` custom hook, which fits perfectly for this case:
```js
const {value: { formula }, error} = useAsync(lens.stateHelperApi, [lens]);
```